src/producer.ts

```diff
@@ -0,0 +1,95 @@
+import { SQS } from 'aws-sdk';
+import { entryFromMessage } from './types';
+const requiredOptions = [
+  'queueUrl'
+];
+
+interface Options {
+  queueUrl?: string;
+  batchSize?: number;
+  sqs?: SQS;
+  region?: string;
+}
+
+interface ProducerInterface {
```

Comment:
We probably don't need this interface, as there is only one class. It's unlikely that we'll need multiple implementations of the Producer at a given time.

Reply:
agreed